experimental/dds/tree2/src/feature-libraries/editable-tree-2/lazyField.ts

```diff
@@ -190,7 +191,196 @@ export abstract class LazyField<TKind extends FieldKindTypes, TTypes extends All
 		);
 	}
 
-	public mapBoxed<U>(
+	public indexOf(searchElement: UnboxNodeUnion<TTypes>, fromIndex = 0): number {
+		return (
+			forEachNode(this[cursorSymbol], (cursor) => {
+				if (cursor.fieldIndex >= fromIndex) {
```

Comment:
Nit: the underlying cursor has the ability to seek to a specific offset.  It should be easy to expose an alternative helper API that leverages this to skip the uninteresting nodes.

Reply:
Oh, great point. I'm assuming you're talking about all the nodes before "fromIndex" - we should definitely skip those via the cursor rather than iterating past them.